packages/builders/src/util/equatable.ts

```diff
@@ -0,0 +1,14 @@
+export interface Equatable<T> {
+	/**
+	 * Whether or not this is equal to another structure
+	 */
+	equals: (other: T) => boolean;
+}
+
+/**
+ * Indicates if an object is equatable or not.
+ * @param maybeEquatable The object to check against
+ */
+export function isEquatable(maybeEquatable: unknown): maybeEquatable is Equatable<unknown> {
```

Comment:
```suggestion
export function isEquatable<T>(maybeEquatable: T): maybeEquatable is Equatable<T> {
```

Or when you'll use isEqualable it'll become unknown.

Reply:
This is inaccurate because it doesn't take `API` types into account.